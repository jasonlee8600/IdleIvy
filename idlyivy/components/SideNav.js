import Link from "next/link";
import { useWeb3React } from "@web3-react/core";
import { getContract } from "../conect/yaleContract";

function SideNav({ page, image, balance, rate, mintable, init}) {
    
    const web3reactContext = useWeb3React();

    async function mint() {

        const YaleContract = getContract(
            web3reactContext.library,
            web3reactContext.account
          );
          
          if(web3reactContext.account != undefined) { 
            if (mintable > 0) {
                try {
                    let mint = await YaleContract.mintCoin();
            
                    await mint.wait();
            
                    init();
                    }
                catch (error) {
                    console.log(error)
                }
            }
            else {
                console.log("Nothing to mint")
            }
            
        }
        else {
            console.log("Not connected")
        }
            
    }

    
    return (
        <div className="flex flex-col items-center hidden h-screen md:flex w-[250px] md:min-w-[250px] lg:min-w-[250px] fixed top-0 gap-2 shadow-2xl bg-gradient-to-b from-[#272727] to-[#797979] outline outline-white outline-[5px]">
            <div className="fixed top-[2%]">
                <Link data-testid="return-home" href="/">
                        <h2 data-testid="IdleIvy" className="text-white font-semibold text-[60px]">
                            IdleIvy
                        </h2>
                </Link>
            </div>

            <div className="fixed top-[15%]">
                <img className="mt=[20px]" src={image}></img>
            </div>

            <div className="fixed top-[35%]">
                <div className="flex flex-col items-center mt-[20px]">
                    <h2 className={`text-4xl text-white`}>
                        Balance:
                    </h2>
                    <h3 className={`text-2xl text-white`}>{balance/10}</h3>
                </div>

                <div className="flex flex-col items-center mt-[25px]">
                    <h2 className={`text-2xl text-white`}>
                        Rate:
                    </h2>
                    <h3 className={`text-xl text-white`}>{rate} tokens/min</h3>
                </div>


                <button type="button" className="bg-[#202B64] mt-[50px] px-[35px] py-[10px] rounded-full outline outline-white outline-[7px]"
                    onClick={mint}> 
                    <div className="flex flex-col items-center">
                        <h2 className={`text-2xl text-white`}>
                            Mint:
                        </h2>
                        <h3 className={`text-xl text-white`}>{mintable/10}</h3>
                    </div>
                </button>
            </div>

            <div className="fixed top-[80%]">
                <div className="flex flex-col items-center">
                    <h2 className="text-white font-semibold text-[30px]">
                        Leaderboard
                    </h2>
                </div>

                <div className="flex flex-col items-center mt-[20px]">
                    <Link href="/">
                        <h2 className="text-white font-semibold text-[30px]">
                            Home
                        </h2>
                    </Link>
                </div>
            </div>

        </div>

    );
}

export default SideNav;
