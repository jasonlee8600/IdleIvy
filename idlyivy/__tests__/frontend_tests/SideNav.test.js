import React from 'react';
import SideNav from '../../components/SideNav'
import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer';

afterEach(() => {
    cleanup(); 
})

describe("SideNav testing", () => {
    test('renders correctly', () => {
        const tree = renderer.create(<SideNav />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    test('Check title is IdleIvy', () => {
        render(<SideNav/>)
        const title = screen.getByTestId('title')
        expect(title.innerHTML).toEqual('IdleIvy')
    });
    
    test('Check balance text', () => {
        render(<SideNav />)
        const text = screen.getByTestId('balance')
        expect(text.innerHTML).toEqual('Balance:')
    });
    
    test('Check rate text', () => {
        render(<SideNav />)
        const text = screen.getByTestId('rate')
        expect(text.innerHTML).toEqual('Rate:')
    });
    
    test('Check leaderboard text', () => {
        render(<SideNav />)
        const text = screen.getByTestId('leaderboard')
        expect(text.innerHTML).toEqual('Leaderboard')
    });
    
    test('Check home text', () => {
        render(<SideNav />)
        const text = screen.getByTestId('home')
        expect(text.innerHTML).toEqual('Home')
    });
});
