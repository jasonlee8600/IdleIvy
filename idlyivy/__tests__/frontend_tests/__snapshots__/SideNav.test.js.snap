// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav testing renders correctly 1`] = `
<div
  className="hidden md:flex flex-col items-center w-[250px] min-w-[250px] h-full bg-gradient-to-b from-[#272727] to-[#797979] gap-[30px] 2xl:gap-[60px] h-screen"
>
  <div
    className="mt-[30px]"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="text-white font-semibold text-[60px]"
        data-testid="title"
      >
        IdleIvy
      </h2>
    </a>
  </div>
  <div
    className=""
  >
    <img
      className="mt=[0px]"
    />
  </div>
  <div
    className=""
  >
    <div
      className="flex flex-col items-center mt-[0px]"
    >
      <h2
        className="text-3xl text-white"
        data-testid="balance"
      >
        Balance:
      </h2>
      <h3
        className="text-2xl text-white"
      >
        NaN
      </h3>
    </div>
    <div
      className="flex flex-col items-center mt-[25px]"
    >
      <h2
        className="text-3xl text-white"
        data-testid="rate"
      >
        Rate:
      </h2>
      <h3
        className="text-2xl text-white"
      >
         tokens/min
      </h3>
    </div>
  </div>
  <div>
    <button
      className="bg-[#202B64] w-40 px-[35px] py-[10px] rounded-full outline outline-white outline-[7px]"
      type="button"
    >
      <div
        className="flex flex-col items-center"
      >
        <h2
          className="text-2xl text-white"
        >
          Connect Wallet!
        </h2>
      </div>
    </button>
  </div>
  <div
    className="mb-[70px]"
  >
    <div
      className="flex flex-col items-center"
    >
      <a
        href="./leaderboard"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <h2
          className="text-white font-semibold text-[30px]"
          data-testid="leaderboard"
        >
          Leaderboard
        </h2>
      </a>
    </div>
    <div
      className="flex flex-col items-center mt-[20px]"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <h2
          className="text-white font-semibold text-[30px]"
          data-testid="home"
        >
          Home
        </h2>
      </a>
    </div>
  </div>
</div>
`;
